### DISCUSSION TITLE
App not working properly

### DISCUSSION THREAD

**Discussion by Chickenshrimps** [KRTirtho/spotube#1375] [Q&A]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

The interface don't load quickly and lags when downloading songs . When I exit the app , the downloaded songs gets deleted automatically 

### Expected Behavior

Smooth interface and working of app 

### Steps to reproduce

1. I opened app
2. It lags

### Operating System

Android 12

### Spotube version

Latest 

### Installation source

GitHub Releases (Binary)

### Additional information

No

### Self grab

- [X] I'm ready to work on this issue!

**Comment by KRTirtho** [comment]
The downloaded tracks are available in the "Library" > "Local Tracks" tab. There's a refresh button that you can use to refresh the track list after download

**Reply by Chickenshrimps** [reply]
I did that . But while downloading songs it lags and when I reopen the app it disappears from the downloaded list 
I just wanté to know that does it support Android 12 if not looking forward for it to be added in the next update 👍

**Reply by KRTirtho** [reply]
While it's downloading, don't quit the app. It'll cancel downloads as currently we don't support downloading in background
